networkx/generators/geometric.py

```diff
@@ -77,34 +118,33 @@ def random_geometric_graph(n, radius, dim=2, pos=None):
 
     References
     ----------
-    .. [1] Penrose, Mathew, Random Geometric Graphs,
-       Oxford Studies in Probability, 5, 2003.
+    .. [1] Penrose, Mathew, *Random Geometric Graphs*,
+           Oxford Studies in Probability, 5, 2003.
+
     """
-    G=nx.Graph()
-    G.name="Random Geometric Graph"
+    G = nx.Graph()
+    G.name = 'random_geometric_graph({}, {}, {})'.format(n, radius, dim)
     G.add_nodes_from(range(n))
+    # If no positions are provided, choose uniformly random vectors in
+    # Euclidean space of the specified dimension.
     if pos is None:
-        # random positions
-        for n in G:
-            G.node[n]['pos']=[random.random() for i in range(0,dim)]
-    else:
-        nx.set_node_attributes(G,'pos',pos)
-    # connect nodes within "radius" of each other
-    # n^2 algorithm, could use a k-d tree implementation
+        pos = {v: [random.random() for i in range(dim)] for v in range(n)}
+    nx.set_node_attributes(G, 'pos', pos)
+    # If no distance metric is provided, use Euclidean distance.
+    if metric is None:
+        metric = euclidean
+    # Join each pair of nodes whose distance is at most `radius`.
+    #
+    # TODO This is an O(n^2) algorithm, a k-d tree could be used
+    # instead.
     nodes = G.nodes(data=True)
-    while nodes:
-        u,du = nodes.pop()
-        pu = du['pos']
-        for v,dv in nodes:
-            pv = dv['pos']
-            d = sum(((a-b)**2 for a,b in zip(pu,pv)))
-            if d <= radius**2:
-                G.add_edge(u,v)
+    G.add_edges_from((u, v) for (u, du), (v, dv) in combinations(nodes, 2)
```

Comment:
Will this add self-loops? The original algorithm does not. I am not sure if it should have.